skyportal/facility_apis/winter.py

```diff
@@ -0,0 +1,405 @@
+import requests
+from requests.auth import HTTPBasicAuth
+from datetime import datetime, timedelta
+from astropy.time import Time
+import urllib
+
+from . import FollowUpAPI
+from baselayer.app.env import load_env
+from baselayer.app.flow import Flow
+from baselayer.log import make_log
+
+from ..utils import http
+
+env, cfg = load_env()
+
+
+WINTER_URL = f"{cfg['app.winter.protocol']}://{cfg['app.winter.host']}"
+if cfg['app.winter.port'] is not None and int(cfg['app.winter.port']) not in [80, 443]:
+    WINTER_URL += f":{cfg['app.winter.port']}"
+
+log = make_log('facility_apis/winter')
+
+WINTER_USERNAME = cfg.get('app.winter.username', None)
```

Comment:
I think this belongs in the allocation secrets, which is where we put such things encrypted.

Reply:
So as Robert explained to me, users that will trigger aren't expected to have a WINTER api key/account, only credentials for a program (which is what I've put in altdata already). So it would make sense to have these credentials (that are used regardless of the program/allocation) available instance-wide.

Reply:
What I can do is have the config credentials, and if they aren't there (so if the instance hasn't been configured to use WINTER by sysadmin), then you can still pass those in the allocation as a fallback?

Reply:
Just to be clear @mcoughlin I'm not saying this is ABSOLUTELY necessary, just that it made sense here, but @robertdstein and I have no problem putting that in the altdata if you think that's better.

Reply:
Personally I think it is more consistent with how we have used this. I don't really like the secrets in the config.

Reply:
done